lib/util/package-rules.ts

```diff
@@ -155,7 +155,7 @@ function matchesRule(
     matchPackagePrefixes.length
   ) {
     if (!depName) {
-      return false;
+      return matchPackagePatterns.length !== 0;
```

Comment:
I don't understand the need for this change. The existing logic seems correct, ie if there's any matchPackageX rule and depName is undefined then it can't match and should return false

Reply:
Yes, that's true.    
The usecase I want to cover here is if there are neither matchPackage rules nor a depName is defined.     
In that case it will simply return false as the logic makes no difference between a user provided `matchPackagePatterns` or the default applied one at https://github.com/renovatebot/renovate/pull/16930/files#diff-e812da1aee40bef2f49c67549d9acf3a805c426089f2adfc60741350ca12391dR63.   
My argument is that in the later should default to match everything, such as in the case of the test. 

This is also behavior for all other matches.